### ISSUE TITLE
textarea character count should support screen readers

### ISSUE THREAD

**Issue by mjuniper** [Esri/calcite-design-system#10492]
**monday.com sync:** #18048700831

### Check existing issues

- [X] I have [checked for existing issues](https://github.com/Esri/calcite-design-system/issues) to avoid duplicates and reviewed the [Accessibility page](https://developers.arcgis.com/calcite-design-system/foundations/accessibility) for guidance.

### Summary

We got an [accessibility violation](https://devtopia.esri.com/dc/hub/issues/11196) on calcite-text-area when LevelAccess audited Hub.

Updates to the character counter (when a maxLength is specified) need to be announced.

### Actual Behavior

Updates to the character count element are not announced by screen readers.

### Expected Behavior

Updates to the character count element should be announced by screen readers.

### Reproduction Sample

https://codepen.io/mjuniper/pen/eYqdNdg?editors=1000

### Reproduction Steps

1. Visit https://codepen.io/mjuniper/pen/eYqdNdg?editors=1000
2. Turn on voiceover
3. Type in the textarea
4. Observe that updates to the character count are not announced.

### Reproduction Version

2.13.0

### Priority impact

impact - p2 - want for an upcoming milestone

### Calcite package

- [X] @esri/calcite-components
- [ ] @esri/calcite-components-angular
- [ ] @esri/calcite-components-react
- [ ] @esri/calcite-design-tokens
- [ ] @esri/eslint-plugin-calcite-components

### Esri team

ArcGIS Hub

**Comment by geospatialem** [comment]
@mjuniper The above is supported in Calcite with JAWS and NVDA as of `1.5.0` (July 2023) via https://github.com/Esri/calcite-design-system/issues/6630. I re-tested and was able to gain context in `2.13.0` with the above example in Windows' JAWS and NVDA. Do you know if an earlier version of Calcite was tested with Level?

**Comment by mjuniper** [comment]
It was definitely an earlier version but I don't know what version.

But at 2.13, VoiceOver does not announce the character counter but it does announce when you are over the character limit.

**Comment by geospatialem** [comment]
Adding a spike to conduct more thorough testing across AT. In coordination with Hub team members it appears the Level test was conducted with Calcite `1.11.0`, so the above fix would have been adopted.

**Comment by geospatialem** [comment]
As mentioned above this only occurs when the user is interacting with the component when the character count is under the specified `maxLength`. In perusing WCAG criteria, this seems to fall under [SC 3.3.5: Help (Level AAA)](https://www.w3.org/WAI/WCAG21/Understanding/help.html), however we do provide the text visually, so we should aim to provide this support to assistive technologies. 

This is also provided in the U.S. Web Design System (USWDS)'s [Character count](https://designsystem.digital.gov/components/character-count/) component.

![image](https://github.com/user-attachments/assets/cbb13d28-d475-4301-8621-7cfbad732a4f)

This effort is low since it is a level AAA, but `text-area` should accommodate providing context when `maxLength` is specified, where:
- [ ] Context is provided when the `maxLength` is specified (currently offered)
- [ ] Context is provided that a `maxLength` is enabled on the component, and the characters remaining should be provided back to assistive technologies

Also sending this request over to design to determine if Calcite should consider only depicting the remaining characters visually, changing the component's current design and behavior.

**Comment by ashetland** [comment]
@geospatialem Unless there's a specific a11y reason to change it, we'd like to keep the current visual design "200/250" which was also approved by the i18n team. That said, is there any concern about a possible mismatch with the AT readout? How would the current implementation be announced to AT users?

**Comment by geospatialem** [comment]
> Unless there's a specific a11y reason to change it, we'd like to keep the current visual design "200/250" which was also approved by the i18n team.

Agreed on this approach 👍🏻 

> That said, is there any concern about a possible mismatch with the AT readout? How would the current implementation be announced to AT users?

@ashetland We could tackle the above in adding additional context through an `aria-label`/`aria-labelledby` to provide context that the component has a maximum character count and/or announce the remaining characters, which would not visually change the component.

**Comment by ashetland** [comment]
Sounds good!
